sktime/decomposition/vampnet.py

```diff
@@ -0,0 +1,662 @@
+            an output activation like softmax which produces a probability distribution over said states.
+        """
+        super().__init__()
+        layers = []
+        if initial_batchnorm:
+            layers.append(nn.BatchNorm1d(units[0]))
+        for fan_in, fan_out in zip(units[:-2], units[1:-1]):
+            layers.append(nn.Linear(fan_in, fan_out))
+            layers.append(nonlinearity())
+        layers.append(nn.Linear(units[-2], units[-1]))
+        if output_nonlinearity is not None:
+            layers.append(output_nonlinearity())
+        self._sequential = nn.Sequential(*layers)
+
+    def forward(self, inputs):
+        return self._sequential(inputs)
+
+
+class VAMPNetModel(Model, Transformer):
+    r"""
+    A VAMPNet model which can be fit to data optimizing for one of the implemented VAMP scores.
+
+    See Also
+    --------
+    VAMPNet : The corresponding estimator.
+    """
+
+    def __init__(self, lobe: nn.Module, lobe_timelagged: Optional[nn.Module] = None,
+                 dtype=np.float32, device=None):
+        r"""Creates a new VAMPNet estimator instance.
+
+        Parameters
+        ----------
+        lobe : torch.nn.Module
+            One of the lobes of the VAMPNet. See also :class:`MLPLobe`.
+        lobe_timelagged : torch.nn.Module, optional, default=None
+            The timelagged lobe. Can be left None, in which case the lobes are shared.
+        dtype : data type, default=np.float32
+            The data type for which operations should be performed. Leads to an appropriate cast within fit and
+            transform methods.
+        device : device, default=None
+            The device for the lobe(s). Can be None which defaults to CPU.
+        """
+        super().__init__()
+        self._lobe = lobe
+        self._lobe_timelagged = lobe_timelagged if lobe_timelagged is not None else lobe
+
+        self._device = device
+        self._dtype = dtype
+
+        if self._dtype == np.float32:
+            self._lobe = self._lobe.float()
+            self._lobe_timelagged = self._lobe_timelagged.float()
+        elif self._dtype == np.float64:
+            self._lobe = self._lobe.double()
+            self._lobe_timelagged = self._lobe_timelagged.double()
+        else:
+            raise ValueError(f"Unsupported type {dtype}! Only float32 and float64 are allowed.")
+
+    def transform(self, data, **kwargs):
+        r""" Transforms a tensor or array or list thereof using the learnt transformation.
+
+        Parameters
+        ----------
+        data : array_like
+            The input data.
+        **kwargs
+            Scikit-learn compatibility.
+
+        Returns
+        -------
+        transform : array_like
+            The featurized data.
+        """
+        self._lobe.eval()
+        self._lobe_timelagged.eval()
+
+        with torch.no_grad():
+            if not isinstance(data, (list, tuple)):
```

Comment:
are you sure there's no case where this does something undesirable?

Reply:
this just treats every singly trajectory as a list of trajectories, the list's length being 1

Reply:
just wanted to make sure there's no way the data could show up in a way where something else happens. like if if it's a np array of trajectories instead of a list.

Reply:
sure you can always find cases which break the implementation because python is not strongly typed - i think it is fair to assume that trajectories == numpy arrays and lists of trajectories == lists of numpy arrays

Reply:
i am cool with that assumption

Reply:
table this convo for later?